### PR TITLE
Simplify WP Photo Directory script and get missing authors

### LINKED ISSUE #322: WP Photo Directory images missing author
## Description

Viewing single results from the WP photo directory, the author name is not present. I am guessing this is a catalog level issue but have not looked into the DB yet.

## Reproduction

Visit https://wordpress.org/openverse/photos/f1d2a27a-054b-4f56-8507-c5bdc9c090d8 and observe the missing author information.

## Screenshots

image

## Environment

 - Device: 
 - OS: 
 - Browser: 
 - Version: 
 - Other info: 

## Resolution

- [ ] 🙋 I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by krysal**
## Fixes

Fixes #322 by @zackkrida 

## Description

This PR simplifies a lot of getting and processing the info for this provider script, as now knowing we can get all the necessary info from just the endpoint that lists photos when adding the `_embed=True` param. This of course includes the author data, that was recently [fixed in the API](https://github.com/WordPress/wporg-mu-plugins/pull/205) to return links with user slugs (publicly accessible) instead of user ids (limited to some users only).

With these changes, the script finishes much faster too ⚡️

### Total images collected

```sql
SELECT COUNT(identifier) FROM image where provider='wordpress';
-- count: 2609
```
### Total images without author
```sql
SELECT COUNT(*) FROM image WHERE provider='wordpress' AND (creator is NULL OR creator = '');
-- count: 0
```

## Testing Instructions

`just test -k wordpress` and/or run the DAG from the Airflow UI and inspect the output loaded into the database.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ (coming soon) ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: openverse_catalog/dags/providers/provider_api_scripts/wordpress.py [hunk 1]
```diff
@@ -264,32 +210,34 @@ def _get_file_info(image_details):
...
 
 
 def _get_title(image):
     if title := image.get("content", {}).get("rendered"):
-        title = html.fromstring(title).text_content()
+        try:
+            title = html.fromstring(title).text_content()
+        except UnicodeDecodeError as e:
+            logger.warning(f"Can't save the image's title ('{title}') due to {e}")
+            return None
```
Comment (krysal) [inline openverse_catalog/dags/providers/provider_api_scripts/wordpress.py:234]: There were some images including emojis in this field that we treat as the title crashing the script. I'm not sure how to handle emojis yet.

**Review by zackkrida — APPROVED** [review]
Looks perfect! Author information is correct and I was able to download the metadata for all WP Photo Directory images (except those with emoji titles, as you mentioned) in 2-3 minutes. 

Here's the count from my CSV file downloaded from MinIO.

```
catalog on  fix/wp_authors +/- [$] via 🐍 v3.9.12
❯ xsv count ~/Downloads/wordpress_image_v001_20220519164607.tsv
2608 
```

And a sample of the author data:

```
catalog on  fix/wp_authors +/- [$] via 🐍 v3.9.12
❯ xsv select 9,10 ~/Downloads/wordpress_image_v001_20220519164607.tsv  | xsv sample 10 | xsv table
Umesh Gupta       https://wordpress.org/photos/author/umesh84
Josh Pollock      https://JoshPress.net
mdburnette        https://mburnette.com
Topher            https://topher1kenobe.com
Topher            https://topher1kenobe.com
Bhavesh           https://wordpress.org/photos/author/bhaveshdesai13
Mark Westguard    https://wsform.com
annezazu          https://nomad.blog
mdburnette        https://mburnette.com
Josh Pollock      https://JoshPress.net
Michael Burridge  https://wordpress.org/photos/author/mburridge
```

Can you make a separate issue for the emoji issue, @krysal?

**Review by obulat — CHANGES_REQUESTED** [review]
It works well for the authors 👍  Nice to see familiar names there, too :)

Could you also add the `filesize` here?
It seems to be available at `wp:featuredmedia.media_details.filesize`. The `full_item` in the test resources has it, I hope other results do to.

**Comment by krysal** [comment]
@obulat Good catch! I had the feeling I was missing something but couldn't tell what exactly 😆 Added in the last commit :) Also with an improvement to get an image not so big, I choose the "2048x2048" in the first place given the resolution I saw in #506 for that size.

**Review by obulat — APPROVED** [review]

File: openverse_catalog/dags/providers/provider_api_scripts/wordpress.py [hunk 2]
```diff
@@ -217,79 +158,105 @@ def _extract_image_data(media_data):
...
+            filetype = Path(filename).suffix.replace(".", "")
 
-def _get_thumbnail_url(image_details):
-    return (
-        image_details.get("media_details", {})
-        .get("sizes", {})
-        .get("thumbnail", {})
-        .get("source_url")
-    )
+        filesize = (
```
Comment (obulat) [inline openverse_catalog/dags/providers/provider_api_scripts/wordpress.py:214]: 🔥 Nice and thorough!

**Pull request merged**